test/bucket_id_test.lua

```diff
@@ -0,0 +1,227 @@
+    t.assert_str_contains(err,
+        'No sharding function specified in DDL schema of space (non-existent-space)')
+    t.assert_equals(id, nil)
+end
+
+function g.test_bucket_id_sharding_key_is_nil()
+    local id, err = ddl.bucket_id('space', nil)
+    t.assert_str_contains(err, 'Sharding key specified for space (space) is nil')
+    t.assert_equals(id, nil)
+end
+
+function g.test_bucket_id_sharding_func_empty_body()
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+    local sharding_func = ''
+    box.space._ddl_sharding_func:insert({
+        'space',
+        box.NULL,
+        sharding_func
+    })
+
+    local id, err = ddl.bucket_id('space', REFERENCE_KEY)
+    t.assert_equals(err,
+        'Failed to execute sharding function for space name (space): attempt to call a nil value')
+    t.assert_equals(id, nil)
+end
+
+function g.test_bucket_id_sharding_func_body()
+    g.schema.spaces.space.sharding_func = {
+        body = helper.sharding_func_body
+    }
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local id, err = ddl.bucket_id('space', REFERENCE_KEY)
+    t.assert_equals(err, nil)
+    t.assert_equals(type(id), 'number')
+    t.assert_equals(id, REFERENCE_BUCKET_ID)
+end
+
+function g.test_bucket_id_sharding_func_user()
+    local sharding_func_name = 'sharding_func'
+    rawset(_G, sharding_func_name, helper.sharding_func)
+    g.schema.spaces.space.sharding_func = sharding_func_name
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local id, err = ddl.bucket_id('space', REFERENCE_KEY)
+    t.assert_equals(err, nil)
+    t.assert_equals(type(id), 'number')
+    t.assert_equals(id, REFERENCE_BUCKET_ID)
+end
+
+function g.test_bucket_id_sharding_func_table_call()
+    local sharding_func_name = 'sharding_func_table_call'
+    local tbl = {}
+    setmetatable(tbl, {
+        __call = helper.sharding_func
+    })
+    rawset(_G, sharding_func_name, tbl)
+    g.schema.spaces.space.sharding_func = sharding_func_name
+
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local id, err = ddl.bucket_id('space', REFERENCE_KEY)
+    t.assert_equals(err, nil)
+    t.assert_equals(type(id), 'number')
+    t.assert_equals(id, REFERENCE_BUCKET_ID)
+end
+
+function g.test_bucket_id_sharding_func_userdata_call()
+    local sharding_func_name = 'sharding_func_userdata_call'
+
+    -- Create a userdata object.
+    local userdata_obj = newproxy(true)
+    local mt = getmetatable(userdata_obj)
+    mt.__call = helper.sharding_func
+    rawset(_G, sharding_func_name, userdata_obj)
+    t.assert_equals(type(userdata_obj), 'userdata')
+
+    -- Set sharding function as a __call method of userdata object.
+    rawset(_G, sharding_func_name, userdata_obj)
```

Comment:
1. The comment does not look relevant to the code.
2. The same rawset() is four lines above.

Reply:
Fixed 1 and 2:
```lua
--- a/test/bucket_id_test.lua
+++ b/test/bucket_id_test.lua
@@ -161,14 +161,13 @@ end
 function g.test_bucket_id_sharding_func_userdata_call()
     local sharding_func_name = 'sharding_func_userdata_call'
 
-    -- Create a userdata object.
+    -- Create a userdata object and set sharding function as a __call method of
+    -- userdata object.
     local userdata_obj = newproxy(true)
     local mt = getmetatable(userdata_obj)
     mt.__call = helper.sharding_func
-    rawset(_G, sharding_func_name, userdata_obj)
     t.assert_equals(type(userdata_obj), 'userdata')
 
-    -- Set sharding function as a __call method of userdata object.
     rawset(_G, sharding_func_name, userdata_obj)
     g.schema.spaces.space.sharding_func = sharding_func_name
```